### DISCUSSION TITLE
Difficulty getting app to work with reverse proxy

### DISCUSSION THREAD

**Discussion by dingdongyo** [Radiokot/photoprism-android-client#107] [General]
I have a photoprism server setup with a Caddy2 reverse proxy and it seems to be working correctly. I can access my photos remotely via a web browser at a URL i set up using noip, and it's using the HTTPS protocol to do it.

I'd like to be able to use this app as well, but I can't get it to connect. When I go to the same URL as the library root URL and use the same username/password, i get a "Library is not accessible: HttpException: HTTP 401" error.

Just to test, I temporarily opened/routed port 2342 on my router to my photoprism server, and appended ":2342" on the URL and that _did_ work, but my understanding is that's very unsafe so I disabled shortly after. I think that ruled out a few possible issues though, which is why i'm assuming it has something to do with the reverse proxy (probably something i haven't configured right).

could i get some help please? i'm learning this stuff as i go and couldn't find any instructions to help myself, apologies if i missed something simple.

**Comment by Radiokot** [comment]
Hi. 
Code 401 means "Unauthorized", so I assume the proxy you use adds extra authorization step which the gallery can't handle.
Please, try opening your library in incognito mode in your browser. If you see not the PhotoPrism login page but something else, the gallery won't work. Alternatively, if you get the PhotoPrism auth page, check the URL in the address bar – if you were redirected, use the final URL.

**Reply by dingdongyo** [reply]
Thanks for the response! I tried going to the URL in an incognito browser as instructed. It loads my photos and works fine. There's no redirect in the address bar. It shows the same URL (with "/library/browse" appended).

**Reply by Radiokot** [reply]
Can you please email me your library URL, so I could check it myself? My address is [oleg@radiokot.com.ua](mailto:oleg@radiokot.com.ua)
